### PR TITLE
Update ApiException thrown error

### PR DESCRIPTION
**Pull request by jason-chan-lf**
Resolves Issue https://github.com/Laserfiche/lf-api-client-core-dotnet/issues/57

Changes
- Removed duplicate `ApiException` and `ProblemDetails` from namespaces `Laserfiche.Api.Client.APIServer` and `Laserfiche.Api.Client.OAuth` and have just one in namespace `Laserfiche.Api.Client`
- Add extra properties (`OperationId`, `ErrorSource`, `ErrorCode`, `TraceId` ) to the `ProblemDetails`
- Remove the `ApiException<T>` type and instead always throw `ApiException` to be more consistent with error handling in the repository client.

### CODE REVIEW

**Comment by jason-chan-lf** [comment]
See https://github.com/Laserfiche/lf-api-client-core-dotnet/pull/58/commits/e4c061f97177cf6bd26f62c5689dfed7ad48f485 to see what changes I made to the template files in the nswag folder.
The template files were copied from the nswag project
https://github.com/RicoSuter/NSwag/tree/master/src/NSwag.CodeGeneration.CSharp/Templates

File: src/ApiException.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        /// <summary>
+        /// Create an <see cref="ApiException"/>. Will try to deserialize the <paramref name="response"/> to a <see cref="Client.ProblemDetails"/>.
+        /// </summary>
+        /// <param name="statusCode">The response status code.</param>
+        /// <param name="headers">The response headers.</param>
+        /// <param name="response">The response body.</param>
+        /// <param name="jsonSerializerSettings">The json serializer settings used to deserialize the response.</param>
+        /// <param name="innerException">The inner exception.</param>
+        /// <returns></returns>
+        public static ApiException Create(int statusCode, IReadOnlyDictionary<string, IEnumerable<string>> headers, string response, JsonSerializerSettings jsonSerializerSettings, Exception innerException)
```
Comment (lf-jason-jia) [inline src/ApiException.cs:41]: Is there any reason that we use `ApiException.Create` instead of `new ApiException`?

Comment (jason-chan-lf) [reply src/ApiException.cs:41]: I wanted to keep the constructor simple (just set properties). This method does extra work like deserializing the response string, so I thought it made sense to use static factory methods to construct the `ApiException`

Comment (lf-jason-jia) [reply src/ApiException.cs:41]: Got it. Thanks for the explanation.

**Review by lf-jason-jia — APPROVED** [review]

**Pull request merged**